libcloud/httplib_ssl.py

```diff
@@ -141,213 +142,99 @@ def _parse_proxy_url(self, proxy_url):
-    def _setup_http_proxy(self):
-        """
-        Set up HTTP proxy.
+    def _setup_verify(self):
+        self.verify = libcloud.security.VERIFY_SSL_CERT
 
-        :param proxy_url: Proxy URL (e.g. http://<host>:3128)
-        :type proxy_url: ``str``
-        """
-        headers = {}
-
-        if self.proxy_username and self.proxy_password:
-            # Include authentication header
-            user_pass = '%s:%s' % (self.proxy_username, self.proxy_password)
-            encoded = base64.encodestring(b(urlunquote(user_pass))).strip()
-            auth_header = 'Basic %s' % (encoded.decode('utf-8'))
-            headers['Proxy-Authorization'] = auth_header
-
-        if hasattr(self, 'set_tunnel'):
-            # Python 2.7 and higher
-            # pylint: disable=no-member
-            self.set_tunnel(host=self.host, port=self.port, headers=headers)
-        elif hasattr(self, '_set_tunnel'):
-            # Python 2.6
-            # pylint: disable=no-member
-            self._set_tunnel(host=self.host, port=self.port, headers=headers)
+    def _setup_ca_cert(self):
+        if self.verify is False:
+            pass
         else:
-            raise ValueError('Unsupported Python version')
-
-        self._set_hostport(host=self.proxy_host, port=self.proxy_port)
-
-    def _activate_http_proxy(self, sock):
-        self.sock = sock
-        self._tunnel()  # pylint: disable=no-member
-
-    def _set_hostport(self, host, port):
-        """
-        Backported from Python stdlib so Proxy support also works with
-        Python 3.4.
-        """
-        if port is None:
-            i = host.rfind(':')
-            j = host.rfind(']')         # ipv6 addresses have [...]
-            if i > j:
-                try:
-                    port = int(host[i + 1:])
-                except ValueError:
-                    msg = "nonnumeric port: '%s'" % (host[i + 1:])
-                    raise httplib.InvalidURL(msg)
-                host = host[:i]
+            if isinstance(libcloud.security.CA_CERTS_PATH, list):
+                if len(libcloud.security.CA_CERTS_PATH) > 1:
+                    warnings.warn('Only 1 certificate path is supported')
+                    print(libcloud.security.CA_CERTS_PATH)
+                self.ca_cert = libcloud.security.CA_CERTS_PATH[0]
             else:
-                port = self.default_port  # pylint: disable=no-member
-            if host and host[0] == '[' and host[-1] == ']':
-                host = host[1:-1]
-        self.host = host
-        self.port = port
+                self.ca_cert = libcloud.security.CA_CERTS_PATH
+
 
+class LibcloudConnection(LibcloudBaseConnection):
+    timeout = None
+    host = None
+    response = None
 
-class LibcloudHTTPConnection(httplib.HTTPConnection, LibcloudBaseConnection):
-    def __init__(self, *args, **kwargs):
+    def __init__(self, host, port, **kwargs):
+        self.host = '{0}://{1}'.format(
+            'https' if port == 443 else 'http',
```

Comment:
this was fixed in #972  and #971 along with a test to do what you just suggested.